tests/test_iqm_sampler.py

```diff
@@ -216,6 +227,42 @@ def test_run_sweep_with_parameter_sweep(adonis_sampler, iqm_metadata, submit_cir
     assert all(isinstance(result, cirq.Result) for result in results)
 
 
+@pytest.mark.usefixtures('unstub')
+def test_run_sweep_abort_job_successful(
+    adonis_sampler, circuit_physical, submit_circuits_default_kwargs, job_id, recwarn
+):
+    client = mock(IQMClient)
+    when(client).submit_circuits(ANY, **submit_circuits_default_kwargs).thenReturn(job_id)
+    when(client).wait_for_results(job_id).thenRaise(KeyboardInterrupt)
+    when(client).abort_job(job_id)
+    when(sys).exit().thenRaise(NotImplementedError)  # just for testing without actually exiting python
```

Comment:
What about simply returning `None`?

Reply:
The reason to use this instead of returning `None` is that otherwise there will be `'NoneType' object is not iterable` error in the last line of `client.run_sweep()` since `measurements` will be `None`.